carbon-design-system/carbon-addons-iot-react
# feat(hierarchy-list): add controlled expandedIds prop

Closes #3037 

**Summary**

- Adds the ability to explicitly control which ids are expanded or collapsed in the HierarchyList. 
- Also adds an `onExpandedChange` callback that passes the array to the user, so they can know what ids are currently expanded

**Change List (commits, features, bugs, etc)**

- adds `expandedIds` prop to control which ids in the list are expanded or collapsed
- adds `onExpandedChange` callback to let the consumer know the current internal state of the expansions.
- added tests to confirm

**Acceptance Test (how to verify the PR)**

- Check https://deploy-preview-3081--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-overflow-menu
- confirm Chicago White Sox is expanded
- Add another team to the expandedIds array knob (comma separated no space--ie. Chicago White Sox,New York Mets)
- Check https://deploy-preview-3081--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-hierarchylist--with-default-expanded-ids
- confirm the white sox and yankees are expanded by default
- Add a different team in the expandedIds array knob and confirm that team is expanded and the white sox and yankees close

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check for errors with defaultExpandedIds story
- check expanding and collapsing on click still works.
- expanded state should only change when the `expandedIds` prop is updated



**Things to look for during review**
